### PR TITLE
[wip] allow static ip and mac with rootless cni network

### PR DESCRIPTION
**Pull request by Luap99**
Make sure we pass the ip and mac address as CNI_ARGS to
the cnitool which is executed in the rootless-cni-infra container.

### CODE REVIEW

File: contrib/rootless-cni-infra/rootless-cni-infra [hunk 1]
```diff
@@ -21,16 +21,18 @@ wait_unshare_net() {
...
-		echo >&2 "Usage: $ARG0 alloc CONTAINER_ID NETWORK_NAME POD_NAME"
+	if [ "$#" -ne 5 ]; then
+		echo >&2 "Usage: $ARG0 alloc CONTAINER_ID NETWORK_NAME POD_NAME IP MAC"
 		exit 1
 	fi
 
 	ID="$1"
 	NET="$2"
 	K8S_POD_NAME="$3"
+	IP="$4"
```
Comment (mheon) [inline contrib/rootless-cni-infra/rootless-cni-infra:34]: Setting a static MAC but not a static IP is allowed - my Bash isn't good enough to know if this supports that, though.

File: libpod/rootless_cni_linux.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ import (
 
 // Built from ../contrib/rootless-cni-infra.
 var rootlessCNIInfraImage = map[string]string{
-	"amd64": "quay.io/libpod/rootless-cni-infra@sha256:304742d5d221211df4ec672807a5842ff11e3729c50bc424ea0cea858f69d7b7", // 3-amd64
+	"amd64": "quay.io/luap99/rootless-cni-infra@sha256:4e9f1e223463a46d9f9b019c0fa8c902494ed34872f75104d985b23812f19683", // 4-amd64
```
Comment (Luap99) [inline libpod/rootless_cni_linux.go:28]: Somebody needs to build the new image and upload it to the libpod repo. After that I can add it here.

File: contrib/rootless-cni-infra/rootless-cni-infra (hunk 1 above)
Comment (Luap99) [reply contrib/rootless-cni-infra/rootless-cni-infra:34]: Yes that works but it requires #8467 for test

File: libpod/rootless_cni_linux.go (hunk 2 above)
Comment (rhatdan) [reply libpod/rootless_cni_linux.go:28]: @TomSweeneyRedHat is this something you can do?

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM
once tests are happy and the image is situated.

**Comment by zhangguanzhang** [comment]
link to https://github.com/containers/podman/pull/8469

**Comment by zhangguanzhang** [comment]
link to https://github.com/containers/podman/issues/7842

**Comment by Luap99** [comment]
Closed in favor of #8585

**Pull request closed without merging**